server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -199,6 +205,27 @@ private RemoteSegmentMetadata readMetadataFile(String metadataFilename) throws I
         }
     }
 
+    // Visible for testing
+    public static void verifyNoMultipleWriters(List<String> mdFiles) {
+        Map<Tuple<Long, Long>, String> nodesByPrimaryTermAndGeneration = new HashMap<>();
```

Comment:
Not sure if only primary term and generation check would be sufficient. We may also need to consider translog generation.

Reply:
I didn't get that. When can two nodes emit  same `primary term - generation ` in metadata files and differing translog gen ?

Reply:
I am not sure. The only time primary term remains same is during primary relocation. 

But we already use `translog generation` in the prefix which is a global counter (for a shard). The idea was, if at all, primary term and generation is same from two different nodes, translog generation should help us to sort the metadata filenames. 

Now, if we fail shard on same primary term and generation for two different nodes, then we maybe failing it without the real issue (as translog generation would be different). If we think this will never happen, then having translog generation in the filename is redundant.

Reply:
> But we already use translog generation in the prefix which is a global counter (for a shard). The idea was, if at all, primary term and generation is same from two different nodes, translog generation should help us to sort the metadata filenames.

We did use it to sort the metadata filenames. But now after we are using `CloseableRetryableRefreshListener` in  https://github.com/opensearch-project/OpenSearch/pull/8683/ and we let ` in-flight remote store operations drain, before we perform the handoff.` ,  the translog generation in the filename is redundant.

Reply:
I have added `translog gen` as well now , just to make sure that we are not failing replica shards on false alarm.